File: Data Structures & Algorithms/maximum-product-subarray/submission-1.java
class Solution {
    public int maxProduct(int[] nums) {
       int curMax=1,curMin=1;

       int res=nums[0];

       for(int i=0;i<nums.length;i++){

          int temp=curMax*nums[i];

          curMax=Math.max(Math.max(nums[i],temp),nums[i]*curMin);

          curMin=Math.min(Math.min(nums[i],temp),nums[i]*curMin);


          res=Math.max(curMax,res);


       }
       return res;
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        freq = {}
        res = []
        for i in nums :

            freq[i] = freq.get(i,0)+1

        for i in nums:

            if freq[i] > len(nums)//3 and i not in res:
                res.append(i)


        return res